### ISSUE TITLE
audioDeviceDidChange is not called when Bluetooth device is disconnected by pressing its power button

### ISSUE THREAD

**Issue by denizmersinlioglu** [aws/amazon-chime-sdk-ios#242]
**Describe the bug**
- `func audioDeviceDidChange(freshAudioDeviceList: [MediaDevice])` is not called when Bluetooth device is disconnected by **shut down the device by pressing its power button.**
-  It works perfectly when we connect - disconnect bluetooth device from iPhone Bluetooth settings - manually.

**To Reproduce**
1. Join to a session (Chime meeting) with or without a `bluetooth audio` device.
2. Lets say you joined to the session  **with** a bluetooth audio device
     2.1. Disconnect the bluetooth device **by pressing its power button** and you **will** observe `audioDeviceDidChange` is called with correct MediaDevice array.
     2.(2x). Connect the same bluetooth device again and you **will** observe `audioDeviceDidChange` is called with correct MediaDevice array.
     2.(2x+1). Disconnect the same bluetooth device again **by pressing its power button** and you **won't** observe `audioDeviceDidChange` is called.
     --> You can repeat the process again and again.
3. Lets say you joined to the session  **without** a bluetooth audio device
     3.(2x+1). Connect a bluetooth device and you **will** observe `audioDeviceDidChange` is called with correct MediaDevice array.
     3.(2x). Disconnect the same bluetooth device again **by pressing its power button** and you **won't** observe `audioDeviceDidChange` is called.
      --> You can repeat the process again and again.

**Expected behavior**
We expect that `audioDeviceDidChange` is called properly with correct MediaDevice array every time we connect - disconnect **(by pressing its power button)** our `bluetooth audio` device 

**Logs**
``` 
// Join to the session with a bluetooth audio device named WH-1000XM3
Chime - Listing Devices:
Chime - Device 0. iPhone Microphone - audioHandset
Chime - Device 1. Built-in Speaker - audioBuiltInSpeaker
Chime - Device 2. WH-1000XM3 - audioBluetooth

// Disconnect bluetooth device - shut down by pressing its power button
Chime - audioDeviceDidChange at timestamp: 1613841263
Chime - Listing Devices:
Chime - Device 0. iPhone Microphone - audioHandset
Chime - Device 1. Built-in Speaker - audioBuiltInSpeaker

// Connect the same bluetooth device again.
Chime - audioDeviceDidChange at timestamp: 1613841279
Chime - Listing Devices:
Chime - Device 0. iPhone Microphone - audioHandset
Chime - Device 1. Built-in Speaker - audioBuiltInSpeaker
Chime - Device 2. WH-1000XM3 - audioBluetooth

// 1613841279 < TimeStamp < 1613841294
// Disconnect bluetooth device - shut down by pressing its power button
// `audioDeviceDidChange` is not called

// Connect the same bluetooth device again.
Chime - audioDeviceDidChange at timestamp: 1613841294
Chime - Listing Devices:
Chime - Device 0. iPhone Microphone - audioHandset
Chime - Device 1. Built-in Speaker - audioBuiltInSpeaker
Chime - Device 2. WH-1000XM3 - audioBluetooth

// 1613841294 < TimeStamp < 1613841309
// Disconnect bluetooth device - shut down by pressing its power button
// `audioDeviceDidChange` is not called

// Connect the same bluetooth device again.
Chime - audioDeviceDidChange at timestamp: 1613841309
Chime - Listing Devices:
Chime - Device 0. iPhone Microphone - audioHandset
Chime - Device 1. Built-in Speaker - audioBuiltInSpeaker
Chime - Device 2. WH-1000XM3 - audioBluetooth

// 1613841309 < TimeStamp < 1613841331
// Disconnect bluetooth device - shut down by pressing its power button
// `audioDeviceDidChange` is not called

// Connect the same bluetooth device again.
Chime - audioDeviceDidChange at timestamp: 1613841331
Chime - Listing Devices:
Chime - Device 0. iPhone Microphone - audioHandset
Chime - Device 1. Built-in Speaker - audioBuiltInSpeaker
Chime - Device 2. WH-1000XM3 - audioBluetooth
```

**Test environment Info (please complete the following information):**
 - Device: **iPhone11**
 - Bluetooth Audio Device: **WH-1000XM3** 
 - OS: **iOS14.4**
 - Version AmazonChimeSDK: **0.15.0**
 - Version AmazonChimeSDKMedia: **0.11.0**

**Comment by denizmersinlioglu** [comment]
**Update:** 
When you power off bluetooth audio device, audio automatically switched to the

-  `Built-in Speaker - audioBuiltInSpeaker`, if you set `AVAudioSession.sharedInstance().setMode(.videoChat)`, 
-  `iPhone Microphone - audioHandset`, if you set `AVAudioSession.sharedInstance().setMode(.voiceChat)`

event if `audioDeviceDidChange` is not called.

**Comment by hokyungh** [comment]
Thank you for reporting! I just wanted to clarify the behavior. Based on my understanding, in the first time when you press power button, you get `audioDeviceDidChange` for the first time, but any following power button disconnection does not invoke that function right? 

One question.
When you turn headphone off and not getting audioDeviceDidChange callback, does list still include Bluetooth?

**Comment by denizmersinlioglu** [comment]
Thanks for your response, 

> Based on my understanding, in the first time when you press power button, you get audioDeviceDidChange for the first time, but any following power button disconnection does not invoke that function right?
- Yes, this is exactly the first scenario: `Lets say you joined to the session with a bluetooth audio device`.

- On the other hand, second scenario: `Lets say you joined to the session without a bluetooth audio device`. In this scenario you would never get `audioDeviceDidChange` when you power off the bluetooth device.

**Comment by denizmersinlioglu** [comment]
For the question, we recently tried it

when we connect bluetooth device -> `audioDeviceDidChange` called
`print(freshAudioDeviceList.map(\.label))` logs ->  ["iPhone Microphone", "Built-in Speaker", "**WH-1000XM3**"] 

after bluetooth device disconnected by powering off, `audioDeviceDidChange` not called.
after an arbitrary time interval, we logged the audioDevices.
`print(audioVideo?.listAudioDevices().map(\.label))` logs -> Optional(["iPhone Microphone", "Built-in Speaker"])

> One question.
> When you turn headphone off and not getting audioDeviceDidChange callback, does list still include Bluetooth?

So the answer is **NO**, it does not include Bluetooth.

**Comment by zhinang-amazon** [comment]
Hi @denizmersinlioglu,
I did some testing with QC35 bluetooth headset, when the headset is powered off, we are not getting `AVAudioSession.routeChangeNotification` system callback with reason `.oldDeviceUnavailable`. This is needed for our SDK to determine that `audioDeviceDidChange` callback is necessary.

See our default implementation [here](https://github.com/aws/amazon-chime-sdk-ios/blob/master/AmazonChimeSDK/AmazonChimeSDK/device/DefaultDeviceController.swift#L75)

We have submitted a feedback to Apple regarding this issue, and will keep this issue updated.
